great_expectations/execution_engine/sqlalchemy_execution_engine.py

```diff
@@ -722,7 +723,9 @@ def get_domain_records(  # noqa: C901, PLR0912, PLR0915
 
             # SQLAlchemy 2.0 deprecated select_from() from a non-Table asset without a subquery.
             # Implicit coercion of SELECT and textual SELECT constructs into FROM clauses is deprecated.
-            if not isinstance(selectable, sa.Table):
+            if not isinstance(selectable, sa.Table) and not isinstance(
+                selectable, Subquery
+            ):
```

Comment:
```suggestion
            if not isinstance(selectable, (sa.Table, Subquery)):
```
I think we can combine these `isinstance()` checks.